srcs/app/utils/SyscallWrap.cpp

```diff
@@ -151,42 +168,46 @@ int SyscallWrap::setsockoptWr(int sockfd,
 							  int level,
 							  int optname,
 							  const void *optval,
-							  socklen_t optlen,
-							  const std::string &file,
-							  const std::string &func,
-							  std::size_t line) {
+							  socklen_t optlen dbgargs()) {
 	int ret;
 	if ((ret = setsockopt(sockfd, level, optname, optval, optlen)) == -1) {
+#ifdef DBG
 		AddDebuggingInfo_(file, "setsockopt", func, line);
+#else
+		throw std::runtime_error(std::strerror(errno));
+#endif
 	}
 	return ret;
 }
 
 int SyscallWrap::bindWr(int sockfd,
 						const struct sockaddr *addr,
-						socklen_t addrlen,
-						const std::string &file,
-						const std::string &func,
-						std::size_t line) {
+						socklen_t addrlen dbgargs()) {
 	int ret;
 	if ((ret = bind(sockfd, addr, addrlen)) == -1) {
+#ifdef DBG
 		AddDebuggingInfo_(file, "bind", func, line);
+#else
+		throw std::runtime_error(std::strerror(errno));
+#endif
 	}
 	return ret;
 }
 
 int SyscallWrap::listenWr(int sockfd,
-						  int backlog,
-						  const std::string &file,
-						  const std::string &func,
-						  std::size_t line) {
+						  int backlog dbgargs()) {
 	int ret;
 	if ((ret = listen(sockfd, backlog)) == -1) {
+#ifdef DBG
```

Comment:
How are `file`, `func` and `line` referenced if you are just passing a macro that has no reference to those variable names? (just trying to understand how it works)

Reply:
The idea is that dbg() expands to , __FILE__, __FUNCTION__, __LINE__ and those in turn expand to <actual_filename>, <actual_function>, <actual_line> in a debugging build, but they expand to nothing in the regular build (if we don't define the DBG macro as a flag passed to the compiler). The SyscallWr:: functions will have entirely different prototypes (function signatures) depending on whether we define DBG or not.

Reply:
nice!